edb/server/protocol/binary.pyx

```diff
@@ -636,6 +668,81 @@ cdef class EdgeConnection:
         if user not in roles:
             raise errors.AuthenticationError('authentication failed')
 
+    def _auth_jwt(self, user):
+        role = self.server.get_roles().get(user)
+        if role is None:
+            raise errors.AuthenticationError('authentication failed')
+
+        if not self._auth_data:
+            raise errors.AuthenticationError(
+                'authentication failed: no authorization data provided')
+
+        header_value = self._auth_data.decode("ascii")
+        scheme, _, encoded_token = header_value.partition(" ")
+        if scheme.lower() != "bearer":
+            raise errors.AuthenticationError(
+                'authentication failed: unrecognized authentication scheme')
+
+        encoded_token = encoded_token.strip()
+        if not encoded_token:
+            raise errors.AuthenticationError(
+                'authentication failed: malformed JWT')
+
+        ekey = self.server.get_jwe_private_key()
+        skey = self.server.get_jws_public_key()
+
+        try:
+            decrypted_token = jwt.JWT(
+                key=ekey,
+                algs=[
+                    "RSA-OAEP-256",
+                    "ECDH-ES",
+                    "A128GCM",
+                    "A192GCM",
+                    "A256GCM",
+                ],
+                jwt=encoded_token,
+            )
+            token = jwt.JWT(
+                key=skey,
+                algs=["RS256", "ES256"],
+                jwt=decrypted_token.claims,
+            )
+        except jwt.JWException as e:
+            raise errors.AuthenticationError(
+                f'authentication failed: {e.args[0]}'
+            ) from None
+        except Exception as e:
+            raise errors.AuthenticationError(
+                f'authentication failed: cannot decode JWT'
```

Comment:
for easier future debug 

```suggestion
                f'authentication failed: cannot decode JWT ({e.args[0]})'
```

Reply:
A better approach would probably be to log the original error, but still be nebulous in the error message returned to the client to avoid accidentally exposing something that shouldn't be exposed.